src/app/src/main/resources/metadata/input-time-series-config-jsonata.txt

```diff
@@ -1,49 +1,53 @@
 (
 $areaNames := $distinct(data.area_name);
 {
-    "data":$map($areaNames, function($v, $i) {(
-        $areaData := data[area_name=$v];
-        [
-            {
-                "name": $v,
-                "showlegend": false,
-                "x": $areaData.time_period,
-                "y": $areaData.value,
-                "xaxis": 'x' & ($i+1),
-                "yaxis": 'y' & ($i+1),
-                "type": "scatter",
-                "line": {
-                    "color": "rgb(51, 51, 51)"
-                }
-            },
-            {
-                "showlegend": false,
-                "x": data[area_name=$v].time_period,
-                "y": $map(data[area_name=$v].value, function($thv, $thi) {
-                       (($thi > 0) and $thv > (1.25 * ($areaData.value)[$thi-1]))
-                       or
-                       (($thi < $count($areaData.value)-1) and ($areaData.value)[$thi+1] > (1.25 * $thv))
-                       or
-                       (($thi > 0) and $thv < (0.75 * ($areaData.value)[$thi-1]))
-                       or
-                       (($thi < $count($areaData.value)-1) and ($areaData.value)[$thi+1] < (0.75 * $thv))
-                       ? $thv : null
-                 }),
-                "xaxis": 'x' & ($i+1),
-                "yaxis": 'y' & ($i+1),
-                "type": "scatter",
-                "line": {
-                    "color": "rgb(255, 51, 51)"
+    "data":(
+        $mapped := $map($areaNames, function($v, $i) {(
+            $areaData := data[area_name=$v];
+            [
+                {
+                    "name": $v,
+                    "showlegend": false,
+                    "x": $areaData.time_period,
+                    "y": $areaData.value,
+                    "xaxis": 'x' & ($i+1),
+                    "yaxis": 'y' & ($i+1),
+                    "type": "scatter",
+                    "line": {
+                        "color": "rgb(51, 51, 51)"
```

Comment:
I wonder if we might need to change the colour for any reason in future, this seems hardcoded.

Reply:
All of this file is chart config, which can be changed at a later date without needing to make code changes. I don't really want to introduce another layer of 'meta-config' somehow. But I will pull out that colour into a jsonata variable so it only has to occur once.